### PR TITLE
sync_diff_inspector: add config item: chunk-thread-count

### LINKED ISSUE #629: sync-diff-inspector has a slow speed of splitting chunks in the case of multiple tables
## Bug Report

sync-diff-inspector has only 3 fixed number of goroutines to split chunks. When there are many tables, this becomes a bottleneck.

1. What did you do?
use sync-diff-inspector to check many small tables.

2. What did you expect to see?
it should be quickly.

3. What did you see instead?
it is slow, so we need to run multiple sync-diff-inspectors at the same time to speed up.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

5. which tool are you using?
sync-diff-inspector

6. what versionof tool are you using (`pump -V` or `tidb-lightning -V` or `syncer -V`)?

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #629 

### What is changed and how it works?
add config item `chunk-thread-count` to adjust the number of goroutines to split chunks.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Review by lichunzhu — APPROVED** [review]

File: sync_diff_inspector/source/chunks_iter.go [hunk 1]
```diff
@@ -31,31 +31,33 @@ type ChunksIterator struct {
...
 	iter := &ChunksIterator{
-		tableAnalyzer: analyzer,
-		TableDiffs:    tableDiffs,
-		chunksCh:      make(chan *splitter.RangeInfo, 64),
-		errCh:         make(chan error, len(tableDiffs)),
-		cancel:        cancel,
+		splitThreadCount: splitThreadCount,
+		tableAnalyzer:    analyzer,
+		TableDiffs:       tableDiffs,
+		chunksCh:         make(chan *splitter.RangeInfo, 30*splitThreadCount),
```
Comment (3pointer) [inline sync_diff_inspector/source/chunks_iter.go]: could you add some comments about magic number 30?

Comment (Leavrth) [reply sync_diff_inspector/source/chunks_iter.go]: done

**Comment by Leavrth** [comment]
Previously, the sync-diff-inspector allocates a WorkerPool with `check-thread-count` goroutines for the comparison process, and split `spliter-thread-count` tables at the same time. When a table uses buckets to generate chunks, a WorkerPool with `check-count-thread` goroutines is allocated to each bucket splitter. So sync-diff-inspector needs about `check-thread-count` + `check-thread-count` * `split-thread-count` connections at most (compare process + bucket-spliter for each table).

Now, bucket-spliters share workpool, that is, sync-diff-inspector needs about `check-thread-count` + `check-thread-count` + `split-thread-count` connections at most.

**Comment by 3pointer** [comment]
> `check-thread-count` * `split-thread-count`

can we unified the `check-thread-count` and `split-thread-count` to one config? or maybe calculate `split-thread-count` from some exists configs.

**Comment by Leavrth** [comment]
now, make `split-thread-count` unexposed, and let it equals `3+check-thread-count/2`.

**Review by 3pointer — APPROVED** [review]

File: sync_diff_inspector/splitter/bucket.go [hunk 2]
```diff
@@ -221,12 +222,13 @@ func (s *BucketIterator) splitChunkForBucket(ctx context.Context, firstBucketID,
 		chunk.InitChunks(chunks, chunk.Bucket, firstBucketID, lastBucketID, beginIndex, s.table.Collation, s.table.Range, bucketChunkCnt)
 		progress.UpdateTotal(s.progressID, len(chunks), false)
 		s.chunksCh <- chunks
+		s.wg.Done()
```
Comment (lichunzhu) [inline sync_diff_inspector/splitter/bucket.go]: if err != nil it seems that s.wg won't Done?
use this as a defer before L214?

**Review by lichunzhu — COMMENTED** [review]
rest LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**